### PR TITLE
Make activities and questions translatable

### PR DESCRIPTION
**Pull request by samanthamjohn**
## WHAT
Allow you to create translations for activities and questions.
## WHY
So that ELL students can read instructions in their own language. 
## HOW
Added a concern called `Translatable` that encapsulates the logic needed to translate a quill model. 

### Notion Card Links
[Activites](https://www.notion.so/quill/Add-translations-for-activity-instructions-26c26760eb564ac1a68d0713f228d2e5?pvs=4)
[Questions](https://www.notion.so/quill/Add-translations-for-Questions-e5d87594978848ed974a4052cc201283?pvs=4)
### What have you done to QA this feature?
Ran translate!  on all the models in the staging console. 

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES

### CODE REVIEW

**Comment by samanthamjohn** [comment]
Note: the test failures will be fixed after https://github.com/empirical-org/Empirical-Core/pull/12013 is merged.

File: services/QuillLMS/app/models/concerns/translatable.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+# app/models/concerns/translatable.rb
```
Comment (brendanshean) [inline services/QuillLMS/app/models/concerns/translatable.rb]: Let's delete this comment.

File: services/QuillLMS/app/models/concerns/translatable.rb [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  extend ActiveSupport::Concern
+
+  SPANISH_LOCALE = "es-la"
+  DEFAULT_LOCALE = SPANISH_LOCALE
+  GENGO_SOURCE = "gengo"
+  OPEN_AI_SOURCE = "open_ai"
+  SOURCES = [OPEN_AI_SOURCE, GENGO_SOURCE]
+
+  included do
+    has_many :translation_mappings, as: :source
```
Comment (brendanshean) [inline services/QuillLMS/app/models/concerns/translatable.rb:13]: `:example_filename` and `:translatable_attribute` values don't appear to change across instances within a class so I think `class_attribute` might make more sense here.


I'd also like to suggest a couple of renames: 
:translatable_attribute -> :translatable_field_name.  # distinguishes column name vs the value
:example_filename -> :translatable_examples.    # prepend translatable to help differentiate it from non concern methods

```
included do
  class_attribute :translatable_field_name, default: nil
  class_attribute :translatable_examples, default: nil
  ...
end
```

Then in a class the configuration seems more natural:

```
class Question < ApplicationRecord
  include Translatable

  translatable_field_name :instructions
  translation_examples "questions.json"
  ...
```

File: services/QuillLMS/app/models/question.rb [hunk 3]
```diff
@@ -163,6 +165,17 @@ def connect_sentence_combining?
     question_type == TYPE_CONNECT_SENTENCE_COMBINING
   end
 
+  # Translatable
+  private def custom_prompt
```
Comment (brendanshean) [inline services/QuillLMS/app/models/question.rb]: This field also feels like a class variable in that it doesn't appear to change across instances, but it's also configuration so maybe create a general config file and pull in the prompt examples with it?

You could create a new folder:  translatable_configs/{activities, questions, ...}.json

where the file would have two top-level keys:

examples: {
...
}
custom_prompt: {
... 
}

Comment (samanthamjohn) [reply services/QuillLMS/app/models/question.rb]: ok, I like that!

File: services/QuillLMS/spec/models/concept_feedback_spec.rb
Comment (brendanshean) [inline services/QuillLMS/spec/models/concept_feedback_spec.rb:1]: Maybe add something about the concern being included:

`it { expect(ConceptFeedback.ancestors).to include(Translatable) }`

**Review by brendanshean — COMMENTED** [review]
Something else I forgot to mention, let's add a couple of tests to check that the concern is included:

it { expect(Activity.ancestors).include(Translatable) } 
it { expect(ConceptFeedback.ancestors).to include(Translatable) }

**Review by brendanshean — APPROVED** [review]
This looks great!  Nice work on extracting out the concern.  I wasn't thinking of yaml when I suggested config but its way easier to parse than JSON.

**Pull request merged**